src/main/java/managers/UpdateManager.java

```diff
@@ -0,0 +1,52 @@
+package managers;
+
+import enums.MoveAttribute;
+import java.util.ArrayList;
+import java.util.Collection;
+import moves.Move;
+
+/**
+ *
+ * @author Jayson
+ */
+public class UpdateManager {
+
+  Collection<Move> moves;
+  Integer currentSize;
+
+  public UpdateManager() {
+    this.moves = new ArrayList<>();
+    this.currentSize = 0;
+  }
+
+  public boolean isReady() {
+    return this.getCurrentSize() >= MoveAttribute.MAX_CAPACITY.getValue();
+  }
+
+  public void reset() {
+    this.setCurrentSize(0);
+    this.setMoves(new ArrayList<>());
+  }
+
+  public void addMove(Move move) {
```

Comment:
It seems this method is designed to add a move to the set although I wonder if the addition of a check for max capacity is needed.  If this move can be added with no check to max cap then there could be an instance where the max is overcome by an addition of a move, this could be the appropriate algorithm however if max capacity is just a classifier on if a moveset is allowed to include more moves before surpassing the cap.

Reply:
The goal is going to have a while loop inside each algorithm's `run()` method. However an extra check inside the `addMove()` method wouldn't hurt it just is somewhat repetitive. 
```java
while(!this.getUpdateManager().isReady()) {
   . . .
   this.getUpdateManager().addMove(move);
}
```

Reply:
Then you are good